fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -373,25 +373,44 @@ instance (Eq a, Display a) => MonadIPFSCluster Server a where
 
       asyncRef <- liftIO $ async do
         Stream.withClientM streamQuery (mkClientEnv ipfsHttpManager url) \event ->
-          runServer cfg $
-            case event of
-              Left clientErr -> do
+          case event of
+            Left clientErr ->
+              runServer cfg do
                 logError $ "🐙😭 Cluster node " <> display url <> " reported streaming client error: " <> display clientErr
                 return $ Left clientErr
 
-              Right ioSource ->
-                liftIO (runExceptT $ Stream.runSourceT ioSource) >>= \case
-                  Left errMsg -> do
+            Right ioSource -> do
+              let
+                withErr errMsg =
+                  runServer cfg do
                     logError $ "🐙😭 Cluster node " <> display url <> " reported generic streaming error: " <> displayShow errMsg
-                    return . Left . ConnectionError . toException $ GenericError errMsg
-
-                  Right vals ->
-                    case List.lastMaybe vals of
-                      Nothing -> do
-                        logError $ "🐙🙉 Cluster node " <> display url <> " did not report any streaming updates."
-                        return . Left . ConnectionError . toException $ NotFound @PinStatus
-
-                      Just final -> do
+                    let err = ConnectionError . toException $ GenericError errMsg
+                    atomically do
+                      latestVar  `writeTVar`  Just (Left err)
+                      resultChan `writeTChan` Left err
+
+                withVal x =
+                  runServer cfg do
+                    logDebug $ "🐙📥 Cluster node " <> display url <> " streamed value: " <> display x
+                    atomically do
+                      latestVar  `writeTVar`  Just (Right x)
+                      resultChan `writeTChan` Right x
```

Comment:
The core problem was probably this ☝️ We didn't actually push anything into the channels, so these would hit the timeout.

Reply:
@walkah lemme know if you'd like a refresher on `TVar`/`TChan`

Reply:
@expede yeah, I actually remember these from the relay.